alembic/operations/ops.py

```diff
@@ -164,6 +164,7 @@ def from_constraint(cls, constraint: Constraint) -> DropConstraintOp:
             "check_constraint": "check",
             "column_check_constraint": "check",
             "table_or_column_check_constraint": "check",
+            "exclude_constraint": "exclude",
```

Comment:
We can't have any hardcoded notion of "exclude" constraint in this file, since this is a PostgreSQL-only concept; not that this would "break" Alembic, but because it establishes that Alembic's architecture is closed to third party dialects (that is, dialect-specific concepts like PostgreSQL ExcludeConstraint act as tests for third party dialects having their own constructs).


these string types are really only used by the MySQL and SQL Server dialects, which have special requirements for their DROP CONSTRAINT syntax.    

Therefore, remove this entry from the dictionary, and instead make the lookup below default to the special string "generic":

     type_=types[constraint.__visit_name__]

becomes:

     type_=types.get(constraint.__visit_name__, "generic")

thanks

Reply:
since we are at it, would this it make sense to expose this dict so that 3rd party extension could add to it?

Reply:
Makes sense.  I've made the change to "generic" instead of "exclude" and pushed an extra commit.